lib/steps/images.js

```diff
@@ -0,0 +1,278 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2019 Joyent, Inc.
+ *
+ * Steps for doing some things with images, like finding an appropriate image
+ * for a new service, or for an upgrade.
+ */
+
+var assert = require('assert-plus');
+var format = require('util').format;
+var vasync = require('vasync');
+var VError = require('verror');
+
+var common = require('../common');
+var errors = require('../errors');
+
+
+// ---- internal support functions
+
+
+// ---- steps
+
+/**
+ * Find an appropriate image for a service.
+ *
+ * Note that "the latest" image is always determined by sorting by
+ * "published_at".
+ *
+ * Input args:
+ * - `args.svcName`: The SAPI service name. This determines the appropriate
+ *   image name.
+ * - `args.channelArg`: Optional. The updates.joyent.com channel to use if
+ *   querying updates.joyent.com. It defaults to the default configured
+ *   channel (per `sdcadm channel`).
+ * - `args.imgArg`: must be one of:
+ *      - "latest": to find the latest available image on updates.joyent.com
+ *      - "current": the find the latest image already installed in the DC's
+ *        local IMGAPI, if any
+ *      - a UUID: to find that particular image
+ *      - a version string: to find that particular image version
+ * - `args.sdcadm`: The SdcAdm object for this run.
+ *
+ * Output args:
+ * - `args.svcImg`: The found image manifest.
+ * - `args.needToDownloadImg`: Set to true if the image is not already in the DC
+ *   and needs to be downloaded from updates.joyent.com.
+ * - `args.channel`: Maybe. If updates.joyent.com was queried (i.e. the image
+ *   was not found locally), then this will be set to `args.channelArg` or
+ *   resolved to the default channel.
+ *
+ * If no matching image can be found, this callback with an error.
+ */
+function findSvcImg(args, cb) {
```

Comment:
I know it's pretty obvious, but shouldn't the function check for `args` being an object and `cb` being a function first. Specially on the `args` case where you could find exceptions of the type `cannot get property args.svcName of undefined`

Reply:
Sure, I can add that.